### PR TITLE
Add mininal test suite to idl2k, remove deprecated ANTLRFileStream

### PR DESCRIPTION
**Pull request by Schahen**
- There's only one test so far that covers the very basic functionality.
- For this particular PR the goal was not to change the code tested at all, that's the reason there's that two new lines in the generated code.
- parameter .. is never used warning fixed
- ANTLRFileStream is deprecated and removed

### CODE REVIEW

File: libraries/tools/idl2k/src/main/kotlin/build.kt [hunk 1]
```diff
@@ -18,16 +19,17 @@ class BuildWebIdl(val mdnCacheFile: File, val srcDir: File) {
             System.exit(1)
         }
 
-        return srcDir.walkTopDown().filter { it.isDirectory || it.extension == "idl" }.asSequence().filter { it.isFile }.toList().sortedBy { it.absolutePath }.fold(Repository(emptyMap(), emptyMap(), emptyMap(), emptyMap())) { acc, e ->
+        return srcDir.walkTopDown().filter { it.isDirectory || it.extension == "idl" }.asSequence().filter { it.isFile }.toList()
+            .sortedBy { it.absolutePath }.fold(Repository(emptyMap(), emptyMap(), emptyMap(), emptyMap())) { acc, e ->
             System.err.flush()
             System.err.println("Parsing ${e.absolutePath}")
-            val fileRepository = parseIDL(ANTLRFileStream(e.absolutePath, "UTF-8"))
+            val fileRepository = parseIDL(CharStreams.fromFileName(e.absolutePath, StandardCharsets.UTF_8))
```
Comment (ilya-g) [inline libraries/tools/idl2k/src/main/kotlin/build.kt]:
```suggestion
            val fileRepository = parseIDL(CharStreams.fromFileName(e.absolutePath, Charsets.UTF_8))
```
`Charsets` object available in `kotlin.text`

File: libraries/tools/idl2k/src/main/kotlin/build.kt [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
-import org.antlr.v4.runtime.ANTLRFileStream
-import kotlin.collections.HashSet
+import org.antlr.v4.runtime.CharStreams
 import java.io.File
 import java.io.IOException
 import java.net.URL
+import java.nio.charset.StandardCharsets
 import java.util.ArrayList
 import java.util.LinkedHashMap
+import kotlin.collections.HashSet
```
Comment (ilya-g) [inline libraries/tools/idl2k/src/main/kotlin/build.kt]: import looks redundant

File: libraries/tools/idl2k/src/main/kotlin/build.kt [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
+import kotlin.collections.contains
+import kotlin.collections.emptyMap
+import kotlin.collections.fold
+import kotlin.collections.forEach
+import kotlin.collections.map
+import kotlin.collections.mapValues
+import kotlin.collections.plus
+import kotlin.collections.reduce
+import kotlin.collections.set
+import kotlin.collections.sortedBy
```
Comment (ilya-g) [inline libraries/tools/idl2k/src/main/kotlin/build.kt]: `kotlin.collections` package is imported by default. Why these imports are required? 😕

File: libraries/tools/idl2k/src/test/kotlin/Idl2kTest.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            "src/test/resources/SomethingNotInCache.idl", """public external open class SomethingNotInCache {
+    open val someReadOnlyParam: dynamic
+    var someWriteableParam: dynamic
+    fun someEmptyMethod(): String
+    fun someMethod(root: dynamic): String
+    fun optionalUsvStringFetcher(name: String): String?
+}
+
+
+"""
```
Comment (ilya-g) [inline libraries/tools/idl2k/src/test/kotlin/Idl2kTest.kt]: Perhaps it would be better to read expected output from a file too

Comment (Schahen) [reply libraries/tools/idl2k/src/test/kotlin/Idl2kTest.kt]: I suggest to have this step after the code will be refactored so  that there will be no two new lines. The thing is in file it's less obvious that there are two empty lines.

**Review by ilya-g — APPROVED** [review]

**Comment by Schahen** [comment]
Merged as squashed and cherry-picked commit

**Pull request closed without merging**